client/p2p/discv5/packet.go

```diff
@@ -0,0 +1,157 @@
+package discv5
```

Comment:
there is a known pattern to solve this problem (specification pattern), that is a bit less rigid, but as is it's easy to understand.